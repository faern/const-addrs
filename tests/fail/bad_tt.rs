use net_macros::{ip, mac, net, sock};

fn main() {
    let a = ip!(192, 168, 1, 1);
    let b = mac!(0xca, 0xfe, 0xc0, 0xff, 0xee, 0x00);
    let c = net!(0x2001, 0xdb0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1);
    let d = sock!(ip!("192.168.1.1"), 300);
}
