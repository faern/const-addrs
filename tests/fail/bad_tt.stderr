error: expected string literal

         = help: Use the string literal for this type's FromStr impl

 --> tests/fail/bad_tt.rs:4:13
  |
4 |     let a = ip!(192, 168, 1, 1);
  |             ^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `ip` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected string literal

         = help: Use the string literal for this type's FromStr impl

 --> tests/fail/bad_tt.rs:5:13
  |
5 |     let b = mac!(0xca, 0xfe, 0xc0, 0xff, 0xee, 0x00);
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `mac` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected string literal

         = help: Use the string literal for this type's FromStr impl

 --> tests/fail/bad_tt.rs:6:13
  |
6 |     let c = net!(0x2001, 0xdb0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1);
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `net` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected string literal

         = help: Use the string literal for this type's FromStr impl

 --> tests/fail/bad_tt.rs:7:13
  |
7 |     let d = sock!(ip!("192.168.1.1"), 300);
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `sock` (in Nightly builds, run with -Z macro-backtrace for more info)
